packages/medusa/src/api-v2/store/payment-collections/route.ts

```diff
@@ -0,0 +1,50 @@
+import { createPaymentCollectionForCartWorkflow } from "@medusajs/core-flows"
+import {
+  ContainerRegistrationKeys,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../types/routing"
+import { StoreCreatePaymentCollectionType } from "./validators"
+
+export const POST = async (
```

Comment:
thought: We should think about how this endpoint can be made more generic, i.e. remove the strict dependency on a `cart_id`. Ideally, you can initiate a payment collection for any type of resource, but this is not important in the first iteration.

Reply:
This was the reason for having the initial payment collection creation go through the `/store/carts` resource.

Reply:
@olivermrbl I agree, but it's better if we have it as a separate endpoint, so if in the future we loosen the requirements (eg. you no longer need to pass a cart_id, but you can pass something else) it won't be a breaking change.